source/Missions/Services/Agents/Messages/AgentMovement.cs

```diff
@@ -0,0 +1,185 @@
+﻿using Common.PacketHandlers;
+using LiteNetLib;
+using Missions.Services.Agents.Packets;
+using ProtoBuf;
+using SandBox.Missions.MissionLogics;
+using System;
+using TaleWorlds.Library;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Messages
+{
+    /// <summary>
+    /// Represents the delta of an <see cref="Agent"/>'s movement between the time it was created and the time it was last updated.
+    /// </summary>
+    [ProtoContract(SkipConstructor = true)]
+    public class AgentMovement : IPacket
+    {
+        /// <summary>
+        /// The <see cref="Vec3"/> representing the direction an <see cref="Agent"/> is looking at.
+        /// </summary>
+        [ProtoMember(1)]
+        public Vec3 LookDirectionDelta = Vec3.Zero;
+
+        /// <summary>
+        /// The <see cref="Vec2"/> representing the direction the <see cref="Agent"/> has input for.
+        /// </summary>
+        [ProtoMember(2)]
+        public Vec2 InputDirectionDelta = Vec2.Zero;
+
+        /// <summary>
+        /// The <see cref="Agent"/>'s <see cref="AgentActionData"/>.
+        /// </summary>
+        [ProtoMember(3)]
+        public AgentActionData ActionDataDelta;
+
+        /// <summary>
+        /// The <see cref="Agent"/>'s <see cref="AgentMountData"/>.
+        /// </summary>
+        [ProtoMember(4)]
+        public AgentMountData MountDataDelta;
+
+        /// <summary>
+        /// The <see cref="Vec3"/> representing the <see cref="Agent"/>'s current position.
+        /// </summary>
+        [ProtoMember(5)]
+        public Vec3 CurrentPosition;
+
+        /// <summary>
+        /// The <see cref="Vec2"/> representing the <see cref="Agent"/>'s current movement direction.
+        /// </summary>
+        [ProtoMember(6)]
+        public Vec2 AgentMovementDirection;
+
+        /// <summary>
+        /// The <see cref="AgentEquipmentData"/> of the <see cref="Agent"/>.
+        /// </summary>
+        [ProtoMember(7)]
+        public AgentEquipmentData AgentEquipmentData;
+
+        /// <summary>
+        /// Guid of the associated <see cref="Agent"/>.
+        /// </summary>
+        [ProtoMember(8)]
+        public Guid AgentId { get; }
+
+        /// <summary>
+        /// The <see cref="PacketType"/>.
+        /// </summary>
+        public PacketType PacketType => PacketType.Movement;
+
+        /// <summary>
+        /// The <see cref="DeliveryMethod"/>.
+        /// </summary>
+        public DeliveryMethod DeliveryMethod => DeliveryMethod.Unreliable;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="agent"></param>
+        /// <param name="guid"></param>
+        public AgentMovement(
```

Comment:
All of these values should initialize to default until an event happens for a given field. Create a constructor using only Guid